audio-worklet/src/bitcrusher-controller.js

```diff
@@ -0,0 +1,145 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ *  Loads sound resources and initiates and configures a Web Audio graph given
+ * GUI controls specified in bitcrusher.html. 
+ * Sound is routed through a bitcrusher wrapper class defined in bitcrusher.js
+ */
+
+/**-----------------------------------------------------------------------------
+ 	Initialize variables and retrieve song from server
+   --------------------------------------------------------------------------**/
+let context = new AudioContext();
+let bitDepth = 8.0; // the precision in bits of the final sample
+let reduction = 10; // bitcrushed buffer will have [reduction] equal samples 
+
+let freq = 440; // default frequency to A4
+let oscillator; // oscillator node
+let mp3Source; // mp3 sound source
+let mp3Buffer; // the Audio Buffer of loaded mp3 track
+let bitcrusher; // script processor wrapper implementing a bit crusher node
+const songs = ["lonely-tentacle.mp3", "revenge.mp3", "it-from-bit.mp3"];
+const folder = "audio/";
+const songUrl = folder + songs[2]; // specified url will be loaded
+
+
+// Retrieve local bitcrusher mp3 file and decode as audio buffer
+fetch(songUrl).then(function(response) { 
+	return response.arrayBuffer();}).then(function(song) { 
+		context.decodeAudioData(song, function(buffer) { // decode as audio buffer
+  		mp3Buffer = buffer;
+    	document.getElementById("startMp3").disabled = false; // enable mp3 button
+  }, function (d) { console.log(d)});
+});
+
+
+/**-----------------------------------------------------------------------------
+***GUI Controls: modify bitcrusher parameters
+***---------------------------------------------------------------------------*/
+
+/**
+ * When oscillator button is clicked, run bitcrushed oscillator at
+ * frequency, bit depth, and sample rate reduction specified in GUI 
+ */
+function testOscillator() {
+  oscillator = new OscillatorNode(context); 
+  bitcrusher = new Bitcrusher(context, {buffersize: 512, inputChannels: 1, 
+  	outputChannels: 1, bitDepth, reduction});
+  oscillator.frequency.value = freq; 
+  
+  // reduce volume of oscillator and connect audio graph
+  let gainNode = new GainNode(context);
+  gainNode.gain.value =0.2;
+  bitcrusher.connectIn(oscillator).connectOut(gainNode);
+  gainNode.connect(context.destination); 
+
+ 
+  oscillator.start();
+
+  // activate controls
+  document.getElementById("stopOscillator").disabled = false;
+  document.getElementById("bdSlider").disabled = false;
+  document.getElementById("srSlider").disabled = false;
+  document.getElementById("startOscillator").disabled = true;
+  document.getElementById("startMp3").disabled = true;
+}
+
+/**
+ * When mp3 button is clicked, play loaded sound source at frequency, 
+ * bit depth, and sample rate reduction specified in GUI  
+ */
+function playMP3() {
```

Comment:
I think this code works for all other decodable file formats. No? Using a specific file format in the function name is a bit odd.